Add UCI Abalone Dataset

mrinalTheCoder opened the pull request:
Adding the Abalone Dataset from UCI Machine Learning repository. The task is to predict the age of abalone using physical measurements. Here is the link to the dataset_info.json: https://gist.github.com/mrinalTheCoder/ec867cff575787cc5d09c8eaf728e5f7

WilliamHYZhang commented on tensorflow_datasets/structured/abalone_test.py:
```diff
@@ -0,0 +1,28 @@
+"""TODO(abalone): Add a description here."""
```
Resolve TODO, please, same for line 12

WilliamHYZhang commented on tensorflow_datasets/structured/abalone_test.py:
```diff
@@ -0,0 +1,28 @@
+  # TODO(abalone):
+  DATASET_CLASS = abalone.Abalone
+  SPLITS = {
+      "train": 3,  # Number of fake train example
```
Get rid of these comments, same for lines 16, 19, 20, 21, 22, 23

mrinalTheCoder commented on tensorflow_datasets/structured/abalone.py:
```diff
@@ -0,0 +1,102 @@
+title = "{UCI} Machine Learning Repository",
+url = "http://archive.ics.uci.edu/ml",
+institution = "University of California, Irvine, School of Information and Computer Sciences" }
+"""
```
This is the correct citation. On the citation policy, this was given, and it said to refer to the bottom of the dataset page for additional citation. There, it just said to refer to the citation policy.

WilliamHYZhang replied:
I was referring to the individual dataset (like the authors of the Abalone dataset specifically) but if you believe this is good then that's fine too.

mrinalTheCoder commented on tensorflow_datasets/structured/abalone.py:
```diff
@@ -0,0 +1,102 @@
+"""
+
+
+def return_same(d):
```
I need to return a string in one case, `return_same` is just doing that. The `return_float` function just casts the values to a float. Refer `titanic.py` in `tensorflow_datasets/structured`.

mrinalTheCoder commented on tensorflow_datasets/structured/abalone.py:
```diff
@@ -0,0 +1,102 @@
+def return_same(d):
+  return d
+
+def return_float(d):
```
Refer the resolution to the above comment :)

mrinalTheCoder commented on tensorflow_datasets/structured/abalone.py:
```diff
@@ -0,0 +1,102 @@
+        yield i, {
+          "age": age_val,
+          "features": {
+            name: FEATURE_DICT[name][1](value)
```
`row` is just a dict with the feature names as the keys with the values. 
`FEATURE_DICT[name]` returns this: `(tf.float32, <function return_float at 0x136f18440>)`
`FEATURE_DICT[name][1]` just gives the function, and by adding `(value)` to that, it calls the function which returns the value in question. Thus the functons are also required.

WilliamHYZhang replied:
Ah ok, thanks.

mrinalTheCoder commented on tensorflow_datasets/structured/abalone.py:
```diff
@@ -0,0 +1,102 @@
+Other measurements, which are easier to obtain, are used to predict the age.
+Further information, such as weather patterns and location
+(hence food availability) may be required to solve the problem.
+"""
```
Fixed in latest commit.

mrinalTheCoder commented on tensorflow_datasets/structured/abalone.py:
```diff
@@ -0,0 +1,102 @@
+class Abalone(tfds.core.GeneratorBasedBuilder):
+  """Abalone age prediction dataset."""
+
+  VERSION = tfds.core.Version('1.0.0')
```
Fixed in latest commit.

mrinalTheCoder commented on tensorflow_datasets/structured/abalone.py:
```diff
@@ -0,0 +1,102 @@
+        name=tfds.Split.TRAIN,
+        gen_kwargs={"file_path": abalone_file}),
+
+        tfds.core.SplitGenerator(
```
Fixed in latest commit.

mrinalTheCoder commented:
The tests ran successfully, here is the output:
`mrinaljain@Mrinals-MacBook-Pro:tensorflow_datasets/structured ‹tfds_abalone*›$ sudo python3 abalone_test.py
Running tests under Python 3.7.6: /usr/local/opt/python/bin/python3.7
[ RUN      ] AbaloneTest.test_baseclass
I0112 22:12:55.565830 4537052608 dataset_builder.py:196] Load pre-computed datasetinfo (eg: splits) from bucket.
[       OK ] AbaloneTest.test_baseclass
[ RUN      ] AbaloneTest.test_download_and_prepare_as_dataset
I0112 22:12:55.568529 4537052608 dataset_builder.py:196] Load pre-computed datasetinfo (eg: splits) from bucket.
Total configs: 0
I0112 22:12:55.571279 4537052608 dataset_builder.py:298] Generating dataset abalone (/tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0)
Downloading and preparing dataset abalone (?? GiB) to /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0...
I0112 22:12:55.637754 4537052608 dataset_builder.py:833] Generating split train
Shuffling and writing examples to /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0.incompleteWO0GI5/abalone-train.tfrecord
  0%|                                                                                       | 0/1 [00:00<?, ? examples/s]I0112 22:12:55.650645 4537052608 tfrecords_writer.py:78] Creating file /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0.incompleteWO0GI5/abalone-train.tfrecord-00000-of-00001
I0112 22:12:55.652634 4537052608 tfrecords_writer.py:175] Done writing /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0.incompleteWO0GI5/abalone-train.tfrecord. Shard lengths: [1]
I0112 22:12:55.653048 4537052608 dataset_builder.py:338] Computing statistics.
Computing statistics...:   0%|                                                                 | 0/1 [00:00<?, ? split/s]I0112 22:12:55.653706 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0.incompleteWO0GI5
2020-01-12 22:12:55.664229: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-01-12 22:12:55.699038: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7f894fc64da0 executing computations on platform Host. Devices:
2020-01-12 22:12:55.699055: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Host, Default Version
                                E0112 22:12:56.173533 4537052608 dataset_info.py:564] Statistics generation doesn't work for nested structures yet       
Computing statistics...: 100%|█████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.92 split/s]
Dataset abalone downloaded and prepared to /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0. Subsequent calls will reuse this data.
I0112 22:12:56.175301 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0
I0112 22:12:56.213206 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0
I0112 22:12:56.284327 4537052608 dataset_builder.py:193] Overwrite dataset info from restored data version.
I0112 22:12:56.286021 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0
I0112 22:12:56.322144 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmptrq0lkib/abalone/1.0.0
I0112 22:12:56.391237 4537052608 dataset_builder.py:196] Load pre-computed datasetinfo (eg: splits) from bucket.
W0112 22:12:56.392739 4537052608 deprecation.py:323] From /usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py:82: TensorFlowTestCase.test_session (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
Instructions for updating:
Use `self.session()` or `self.cached_session()` instead.
Total configs: 0
I0112 22:12:56.395328 4537052608 dataset_builder.py:298] Generating dataset abalone (/tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0)
Downloading and preparing dataset abalone (?? GiB) to /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0...
I0112 22:12:56.396460 4537052608 dataset_builder.py:833] Generating split train
Shuffling and writing examples to /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0.incompleteWO0GI5/abalone-train.tfrecord
  0%|                                                                                       | 0/1 [00:00<?, ? examples/s]I0112 22:12:56.402662 4537052608 tfrecords_writer.py:78] Creating file /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0.incompleteWO0GI5/abalone-train.tfrecord-00000-of-00001
I0112 22:12:56.403136 4537052608 tfrecords_writer.py:175] Done writing /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0.incompleteWO0GI5/abalone-train.tfrecord. Shard lengths: [1]
I0112 22:12:56.403371 4537052608 dataset_builder.py:338] Computing statistics.
Computing statistics...:   0%|                                                                 | 0/1 [00:00<?, ? split/s]I0112 22:12:56.403829 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0.incompleteWO0GI5
                                E0112 22:12:56.577333 4537052608 dataset_info.py:564] Statistics generation doesn't work for nested structures yet       
Computing statistics...: 100%|█████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  5.74 split/s]
Dataset abalone downloaded and prepared to /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0. Subsequent calls will reuse this data.
I0112 22:12:56.579442 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0
I0112 22:12:56.625580 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0
I0112 22:12:56.756598 4537052608 dataset_builder.py:193] Overwrite dataset info from restored data version.
I0112 22:12:56.758031 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0
I0112 22:12:56.802086 4537052608 dataset_builder.py:434] Constructing tf.data.Dataset for split train, from /tmp/abalone_testjzb1es5q/tmpko58653q/abalone/1.0.0
[       OK ] AbaloneTest.test_download_and_prepare_as_dataset
[ RUN      ] AbaloneTest.test_info
I0112 22:12:56.942656 4537052608 dataset_builder.py:196] Load pre-computed datasetinfo (eg: splits) from bucket.
[       OK ] AbaloneTest.test_info
[ RUN      ] AbaloneTest.test_registered
I0112 22:12:56.944444 4537052608 dataset_builder.py:196] Load pre-computed datasetinfo (eg: splits) from bucket.
[       OK ] AbaloneTest.test_registered
[ RUN      ] AbaloneTest.test_session
[  SKIPPED ] AbaloneTest.test_session
----------------------------------------------------------------------
Ran 5 tests in 1.841s

OK (skipped=1)`

mrinalTheCoder commented:
Are there any changes to be made?
If not, please review
Thanks

mrinalTheCoder commented:
Hey, it has been a long time since the dataset was put for review. If there are any improvements/errors, please let me know. If not, please merge :)
